pennylane/tape/qnode.py

```diff
@@ -441,15 +441,18 @@ def construct(self, args, kwargs):
         # pylint: disable=protected-access
         obs_on_same_wire = len(self.qtape._obs_sharing_wires) > 0
         ops_not_supported = any(
-            not self.device.supports_operation(op.name) for op in self.qtape.operations
+            isinstance(op, qml.tape.QuantumTape)  # nested tapes must be expanded
+            or not self.device.supports_operation(op.name)  # unsupported ops must be expanded
+            for op in self.qtape.operations
         )
 
-        # expand out the tape, if any operations are not supported on the device or multiple
-        # observables are measured on the same wire
+        # expand out the tape, if nested tapes are present, any operations are not supported on the
+        # device, or multiple observables are measured on the same wire
         if ops_not_supported or obs_on_same_wire:
             self.qtape = self.qtape.expand(
                 depth=self.max_expansion,
-                stop_at=lambda obj: self.device.supports_operation(obj.name),
+                stop_at=lambda obj: not isinstance(obj, qml.tape.QuantumTape)
+                and self.device.supports_operation(obj.name),

```

Comment:
The `not isinstance(obj, qml.tape.QuantumTape)` instruction basically tells the QNode to _flatten_ the tape if is consists of nested tapes.

This is part of the logic that is due to be moved to the device.

Reply:
What can be the type of a nested tape? Any iterable?

Reply:
the type of a nested tape will be `QuantumTape()` 🙂 This says 'expand the tape, but stop at anything that isn't a QuantumTape'

Reply:
In the batch tape prototype I implemented a recursive expand function...in the long run we should rewrite the tape one as well I think?

Reply:
The current `expand()` function is recursive 🙂